File: HoYa.Web/src/@models/item.ts
import { Item } from "@entities/item";
import { FolderFile } from "@entities/entity";

export class ItemDetail {
    id: string;
    photo: string;
    value: string;
    deletable: boolean;
    status: string;
    description: string;
}

export class ItemSave {
    item: Item;
    categoryValues: string[];
    photos: FolderFile[];
}

export class ItemUpload {
    code: string;
    value: string;
    description: string;
}

export class ItemSelect {
    id: string;
    code: string;
    value: string;
}

export class ItemGrid {
    item: Item;
    recipeDraftQuantity: number;
    recipeQuantity: number;
    inventoryQuantity: number;
    photoQuantity: number;
}

export class ItemAttributeCheckbox {
    _checked: boolean
    value: string;
}

export class ItemList {
    id: string;
    photo: string;
    value: string;
    description: string;
    _checked: boolean;
    deletable: boolean;
}

File: HoYa/Models/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using HoYa.Entities;

namespace HoYa.Models
{
    public class ItemAttributesList
    { 

    }
    public class ItemSave
    {
        public Item Item { get; set; }
        public ICollection<string> CategoryValues { get; set; } = new HashSet<string>();

        public ICollection<FolderFile> Photos { get; set; } = new HashSet<FolderFile>();
    } 
    public class ItemSelect
    {
        public Guid Id { get; set; }
        public string Code { get; set; }
        public string Value { get; set; }
        public string Photo { get; set; }
    } 
    public class ItemList
    {
        public Guid Id { get; set; }
        public string Value { get; set; }
        public string _photo { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
        public bool? _deletable { get; set; }
    }
    public class ItemDetail
    {
        public Guid Id { get; set; }
        public string Value { get; set; }
        public string Photo { get; set; }
        public string Status { get; set; }
        public bool? Deletable { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
    } 
    public class ItemGrid
    {
        public Item Item { get; set; }
        public int RecipeDraftQuantity { get; set; }
        public int RecipeQuantity { get; set; }

        public int InventoryQuantity { get; set; }

        public int PhotoQuantity { get; set; }
    }
     
}